packages/enzyme/src/Utils.js

```diff
@@ -242,3 +243,42 @@ export function cloneElement(adapter, el, props) {
     { ...el.props, ...props },
   );
 }
+
+export function spyMethod(instance, methodName) {
+  let lastReturnValue;
+  const originalMethod = instance[methodName];
+  const hasOwn = has(instance, methodName);
+  let descriptor;
+  if (hasOwn) {
+    descriptor = Object.getOwnPropertyDescriptor(instance, methodName);
+  }
+  Object.defineProperty(instance, methodName, {
+    configurable: true,
+    enumerable: false,
```

Comment:
shouldn't this be `!descriptor || !!descriptor.enumerable`?

Reply:
But shouldn't an instance method be enumerable?

Reply:
the replacement should be enumerable, or not, to match the one it's spying on